Discourage IncludeAllContentForSelfExtract in single-file docs (#55192)
dotnet/docs: docs/core/deploying/single-file/overview.md

To embed those files for extraction and get one output file, set the property `IncludeNativeLibrariesForSelfExtract` to `true`.

> Specifying `IncludeAllContentForSelfExtract` extracts all files, including the managed assemblies, before running the executable. This may be helpful for rare application compatibility problems. We don't recommend this option: it's a .NET Core 3.1 compatibility mode and may be removed in a future release.

Reviewer: ```suggestion
Specifying `IncludeAllContentForSelfExtract` extracts all files, including the managed assemblies, before running the executable. This may be helpful for rare application compatibility problems. This mode is not recommended: it's a .NET Core 3.1 compatibility mode and might be removed in a future release.
```
